src/Streamly/Streams/StreamK.hs

```diff
@@ -798,6 +859,22 @@ sequence m = go (toStream m)
             yieldk ma r = unStream (toStream $ ma |: go r) st stp sng yld
          in (unStream m1) (rstState st) stp single yieldk
 
+-------------------------------------------------------------------------------
+-- Inserting
+-------------------------------------------------------------------------------
+
+{-# INLINE intersperseM #-}
+intersperseM :: (IsStream t, Monad m) => m a -> t m a -> t m a
+intersperseM a m = fromStream $ prependingStart (toStream m)
+    where
+    prependingStart m1 = Stream $ \st stp sng yld ->
+        let yieldk i x = yld i $ go x
+         in unStream m1 st stp sng yieldk
+    go m2 = Stream $ \st stp _ yld ->
+        let single i = a >>= \y -> yld y $ yield i
+            yieldk i x = a >>= \y -> yld y $ cons i $ go x
```

Comment:
`intersperseM` is a monadic function therefore should have concurrency capability. For that you need to make `single` and `yieldk` concurrent, see mapM implementation as an example. Construct a concurrent stream using `|:` and use `unStream` to run it e.g. `unStream (a |:  return i |: go x)`...